ya_gpt_bot/bot_config/handlers/chat.py

```diff
@@ -117,3 +120,44 @@ async def text_generation_request(  # pylint: disable=too-many-arguments
         await messages_service.save_message(result.message_id, message.message_id, message.chat.id, response, True)
     except ya_exc.GenerationTimeoutError:
         await message.reply(responses.timeout_error)
+
+
+@chat_messages_router.message(Command("digest"))
+async def digest_request(
+    message: Message,
+    conversation_service: ConversationService,
+    gpt_client: GPTClient,
+):
+    """command for triggering chat digest"""
+    # around 150 symbols
+    prompt_init = (
+        "Сделай выжимку происходившего в чате используя"
+        + "сsv лог с тремя колонками: от кого, кому, текст сообщения."
+        + " Сообщения идут в хронологическом порядке."
+    )
+    chat_id = message.chat.id
+    messages = await conversation_service.get_messages_within_context(chat_id, CONTEXT_LENGTH - len(prompt_init) - 1)
+    if not messages:
+        await message.bot.send_message(chat_id, text="Что-то пошло не так - ни одно сообщение не попало в контекст")
+        return
+    messages_joined = "\n".join(messages)
+    prompt = f"""{prompt_init}
+{messages_joined}"""
+    try:
+        model_response = await gpt_client.request(prompt, creativity_override=0.0)
```

Comment:
Shouldn't this whole part be inside `ConversationService.get_digest(messages)`?
It seems wrong to hardcode initial prompt inside a command handler

Reply:
Moved prompt creation inside the service. Left gpt_client call in handler though, to follow the style of the other handlers (since we have only engine dependency in the service)
> It seems wrong to hardcode initial prompt inside a command handler
Do you mean that it should be set up in the config?

Reply:
Service is fine already, it would be easier to change prompt anyway if needed